fcrepo-clients/src/main/java/edu/unc/lib/dl/fcrepo4/RepositoryObject.java

```diff
@@ -74,21 +74,29 @@ public URI getUri() {
 	 * @throws FedoraException
 	 */
 	public Model getModel() throws FedoraException {
-		if (model == null) {
-			dataLoader.loadModel(this);
-		}
-
+		dataLoader.loadModel(this);
 		return model;
 	}
 
 	/**
+	 * Return true if the model for this object has been populated
+	 * 
+	 * @return
+	 */
+	public boolean hasModel() {
+		return model != null;
+	}
+
+	/**
 	 * Set the model
 	 * 
 	 * @param model
 	 * @return
 	 */
 	public void setModel(Model model) {
 		this.model = model;
+		// Clear the cached types list
```

Comment:
I switched it to storeModel() to make it clear that the method wouldn't behave like a bean property, but also so it wouldn't feel a little weird when called before a value had been set.  (Un)happy compromise?